src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -77,6 +77,49 @@ public static byte[] Keccak256(byte[] data)
             return data.Keccak256();
         }
 
+        /// <summary>
+        /// Verifies that a digital signature is appropriate for the provided key and message using the ECDSA algorithm.
+        /// </summary>
+        /// <param name="message">The signed message.</param>
+        /// <param name="signature">The signature to be verified.</param>
+        /// <param name="curveHash">A pair of the curve to be used by the ECDSA algorithm and the hasher function to be used to hash message.</param>
+        /// <returns><see langword="true"/> if the signature is valid; otherwise, <see langword="false"/>.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 10)]
+        public static ECPoint[] ECrecover(byte[] message, byte[] signature, NamedCurveHash curveHash)
+        {
+            try
+            {
+                var ch = s_curves[curveHash];
+                var messageHash =
+                   ch.Hasher == Hasher.SHA256 ? message.Sha256() :
+                   ch.Hasher == Hasher.Keccak256 ? message.Keccak256() :
+                   throw new NotSupportedException(nameof(ch.Hasher));
+
+                switch (curveHash)
+                {
+                    case NamedCurveHash.secp256k1Keccak256:
+                    case NamedCurveHash.secp256k1SHA256:
+                        {
+                            return Crypto.ECRecover(ECCurve.Secp256k1, signature, messageHash,
+                                // TODO: only accept 65?
```

Comment:
64bytes signatures MAY store a parity bit in the first bit of the `s` value which is always 0. 
From what I deduce, you assume the parity value to be 0 if the size is 64 so checking the first bit of `s` will not change the current behavior if this is not a compressed signature (i.e. parity will always be 0) But in case it is a compressed ERC-2098 signature, it will be supported out-of-the-box

Note: this is predicated on the assumption that N3 contracts might need to verify signatures that originate in NEOX where EIP standards might apply  (eg: someone migrates a platform written for Ethereum to NeoX)
for reference, this is the compressed signature standard I am referring to:  https://eips.ethereum.org/EIPS/eip-2098